lib/main.d.ts

```diff
@@ -30,6 +30,16 @@ export interface BusboyConfig {
      * @default 'utf8'
      */
     defCharset?: string | undefined;
+    /**
+     * Detect if a Part is a file.
+     * 
+     * By default a file is detected if contentType 
+     * is application/octet-stream or fileName is not
+     * undefined.
+     * 
+     * Modify this to handle e.g. Blobs.
+     */
+    isPartAFile?: ((fieldName?: string | undefined, contentType?: string | undefined, fileName?: string | undefined) => boolean) | undefined;
```

Comment:
`fieldName?: string | undefined`
should be the same as `fieldName?: string`
(or in this case)
`fieldName: string | undefined`

I would go with: `fieldName: string | undefined`, because the args will be passed to the function, but they may be undefined. In contrast to a function where some args might be omitted.
The config itself may be omitted, so `?` is necessary here.

```suggestion
    isPartAFile?: ((fieldName: string | undefined, contentType: string | undefined, fileName: string | undefined) => boolean);
```

I think for consistency it would also be fine to use the optional operator everywhere:
```suggestion
    isPartAFile?: ((fieldName?: string, contentType?: string, fileName?: string) => boolean);
```

Reply:
i think the string|undefined is better describing the function then making them with optional operator. The parameters are not optional, but can be undefined.

Adapted the suggestions